hw3/01-characters.js

```diff
@@ -1,4 +1,75 @@
-// url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const url = "https://thronesapi.com/api/v2/Characters";
 
-fetch(url);
+let list = document.querySelector("#list");
+
+const fetchData = (url) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const response = await fetch(url);
+      const data = await response.json();
+      resolve(data);
+    } catch (error) {
+      reject(error);
+    }
+  });
+};
+
+const createCharacterElements = (character) => {
+  let col = document.createElement("div");
+  col.classList.add("col-3", "p-0", "m-2");
+
+  let figure = document.createElement("figure");
+  figure.classList.add("figure", "h-100");
+
+  figure.addEventListener("mouseover", () => {
+    figure.style.backgroundColor = "darkBlue";
+    figure.style.color = "white";
+  });
+
+  figure.addEventListener("mouseout", () => {
+    figure.style.backgroundColor = "";
+    figure.style.color = "";
+  });
+
+  let img = document.createElement("img");
+  img.classList.add("figure-img", "img-fluid");
+  img.src = character.imageUrl;
+  img.alt = character.image;
```

Comment:
The `image` field makes the alt text the image's file name (i.e. `sam.jpg`), which is not very helpful for accessibility.

I recommend using the characters' name (from `fullName`) to write a short description string.

Reply:
Yes, you're right. The only reason I wrote it so short was because VS was giving me a warning! saying the alt value should not include the words "a picture of" . I wanted to get rid of the warning!